services/productService.js

```diff
@@ -0,0 +1,226 @@
+
+const addProduct = async (req, res, next) => {
+  const userId = req.user.id;
+
+  const {
+    category_id,
+    product_name,
+    price,
+    product_description,
+    product_quantity,
+    images,
+  } = req.body;
+
+  const productData = {
+    user_id: userId,
+    category_id,
+    product_name,
+    price,
+    product_description,
+    product_quantity,
+  };
+
+  const addProduct = await product.create(productData);
+
+  if (images && images.length > 0) {
+    await addImages(addProduct.id, images);
+  }
+
+  if (addProduct) {
+    logger.info(`Product ${Messages.ADD_SUCCESS}`);
+    next(
+      new GeneralResponse(
+        `Product ${Messages.ADD_SUCCESS}`,
+        StatusCodes.OK,
+        undefined,
+        RESPONSE_STATUS.SUCCESS,
+      ),
+    );
+  }
+};
+
+const updateProduct = async (req, res, next) => {
+  const id = req.params.id;
+  const findId = await product.findOne({ where: { id } });
+
+  const { product_name, price, product_description, product_quantity, images } =
+    req.body;
+
+  const updateProductData = {
+    product_name,
+    price,
+    product_description,
+    product_quantity,
+    images,
+  };
+
+  if (findId) {
+    const updateProduct = await product.update(updateProductData, {
+      where: { id },
+    });
+
+    if (images && images.length > 0) {
+      await productImage.destroy({ where: { product_id: id } });
+      await addImages(id, images);
+    }
+
+    if (updateProduct) {
+      logger.info(`Product ${Messages.UPDATE_SUCCESS}`);
+      next(
+        new GeneralResponse(
+          `Product ${Messages.UPDATE_SUCCESS}`,
+          StatusCodes.ACCEPTED,
+          undefined,
+          RESPONSE_STATUS.SUCCESS,
+        ),
+      );
+    }
+  } else {
+    logger.error(`Product ${Messages.NOT_FOUND}`);
+    next(
+      new GeneralError(
+        `Product ${Messages.NOT_FOUND}`,
+        StatusCodes.NOT_FOUND,
+        undefined,
+        RESPONSE_STATUS.ERROR,
+      ),
+    );
+  }
+};
+
+const deleteProduct = async (req, res, next) => {
+  const id = req.params.id;
+  const findId = product.findOne({ where: { id } });
+
+  if (findId) {
+    const deleteProduct = await product.update(
+      { is_deleted: true },
+      { where: { id } },
+    );
+
+    if (deleteProduct) {
+      logger.info(`Product ${Messages.DELETE_SUCCESS}`);
+      next(
+        new GeneralResponse(
+          `Product ${Messages.DELETE_SUCCESS}`,
+          StatusCodes.OK,
+          undefined,
+          RESPONSE_STATUS.SUCCESS,
+        ),
+      );
+    }
+  } else {
+    logger.error(`Product ${Messages.NOT_FOUND}`);
+    next(
+      new GeneralError(
+        `Product ${Messages.NOT_FOUND}`,
+        StatusCodes.NOT_FOUND,
+        undefined,
+        RESPONSE_STATUS.ERROR,
+      ),
+    );
+  }
+};
+
+const deleteProductImage = async (req, res, next) => {
+  const id = req.params.id;
+  const findId = productImage.findOne({ where: { id } });
```

Comment:
same as here

Reply:
in product_image table there no is_deleted field that's why I have not done that